src/H5.c

```diff
@@ -914,7 +930,8 @@ H5check_version(unsigned majnum, unsigned minnum, unsigned relnum)
             disable_version_check = (unsigned int)HDstrtol(s, NULL, 0);
     }
 
-    if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum || H5_VERS_RELEASE != relnum) {
+    /* H5_VERS_MAJOR and H5_VERS_MINOR must match */
+    if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum) {
```

Comment:
Knowing that the compatibility checker considers new binaries compatible with older binaries if no symbols are removed nor signatures changed in the newer binary, it seems that linking an application to a newer binary  in the same major.minor release series should be possible without setting an environment variable, unless there is a known problem with a particular release.  I think this PR achieves that, but perhaps also allows linking to a binary from an older version in the series without setting the environment variable.  Should it be incompatible if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum || H5_VERS_RELEASE > relnum) here?